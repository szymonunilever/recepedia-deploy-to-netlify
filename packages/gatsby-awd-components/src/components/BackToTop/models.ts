import { ReactNode } from 'react';
import { ButtonViewType } from '../Button';
import { UnileverLibraryComponent } from '../../models';

export interface BackToTopProps
  extends UnileverLibraryComponent<AppContent.BackToTopContent> {
  Icon?: JSX.Element;
  children?: ReactNode | ReactNode[];
  viewType?: ButtonViewType;
  hideTopPositionPx?: number;
}

export interface CustomEventTarget {
  scrollingElement: HTMLElement | null;
}
