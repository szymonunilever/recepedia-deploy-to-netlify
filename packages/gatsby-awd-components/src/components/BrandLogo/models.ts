export interface BrandLogoProps {
  className?: string;
  brand?: string;
  linkTo?: string;
}

