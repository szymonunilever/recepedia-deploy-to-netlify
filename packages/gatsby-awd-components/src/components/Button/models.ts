import { ReactNode } from 'react';
import { Icon } from '../../models';

export interface ButtonCallback {
  // @ts-ignore
  (...arg): void;
}
export enum ButtonViewType {
  classic,
  icon,
}

export interface ButtonProps {
  className?: string;
  Icon?: Icon;
  IconSelected?: Icon;
  content?: AppContent.CTAContent;
  role?: string;
  isDisabled?: boolean;
  isSelected?: boolean;
  onClick?: ButtonCallback;
  onKeyDown?: (e: React.KeyboardEvent<HTMLElement>) => void;
  children?: ReactNode | ReactNode[];
  viewType?: ButtonViewType;
  hidden?: boolean;
  isToggle?: boolean;
  type?: 'button' | 'submit' | 'reset';
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  attributes?: Record<string, any>;
}
