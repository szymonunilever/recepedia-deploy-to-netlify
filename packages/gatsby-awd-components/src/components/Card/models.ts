import { ReactChild, ReactElement } from 'react';
import { Icon, RatingAndReviewsProvider, UnileverLibraryComponent } from '../../models';
import { ButtonProps } from '../Button';
import { RatingProps } from '../Rating';

export interface CardProps {
  key: string;
  content: CardContentProps;
  idPropertyName: string;
  ratingWidget?: ReactElement<RatingProps>
  imageSizes: string;
  className?: string;
  brand?: ReactChild;
  children?: ReactElement<ButtonProps> | ReactElement<ButtonProps>[];
}

export interface CardContentProps {
  fields:{
    slug: string
  };
  name?: string;
  title?: string;
  localImage?: Internal.LocalImage;
  images?: Internal.LocalImage[];
  averageRating?: number;
  [key: string]: any;
}

interface CardWrapper {
  key: string;
  children: ReactElement<CardProps>;
  ratingProvider: RatingAndReviewsProvider;
}

export interface RecipeCardWrapperProps extends CardWrapper{}

export interface ProductCardWrapperProps extends CardWrapper{
  mainImageIndex?: number;
}
