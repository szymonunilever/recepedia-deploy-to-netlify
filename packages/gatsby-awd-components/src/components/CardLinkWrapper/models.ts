import { ReactElement } from 'react';
import { CardProps, ProductCardWrapperProps, RecipeCardWrapperProps } from '../Card';
import { RecipeCardProps } from '../RecipeCard';

export interface CardLinkWrapperProps {
  key: string;
  isExternal?: boolean;
  title: string;
  slug: string;
  target?: string;
  href?: string;
  rel?: string;
  to?: string;
  'aria-label'?: string;
  className?: string;
  children: ReactElement<RecipeCardProps|CardProps|RecipeCardWrapperProps|ProductCardWrapperProps>;
}
