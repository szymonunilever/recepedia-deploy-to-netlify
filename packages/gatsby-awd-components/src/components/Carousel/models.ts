import { ReactChild, ReactElement } from 'react';
import { Icon } from '../../models';
import { RecipeCardProps } from '../RecipeListing/partials/models';

export interface CreateElement {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (itemData: any): ReactElement<RecipeCardProps> | undefined;
}

export interface BreakpointProps {
  width: number;
  switchElementsBelowBreakpoint: number;
  switchElementsAfterBreakpoint: number;
  visibleElementsBelowBreakpoint: number;
  visibleElementsAboveBreakpoint: number;
}

export interface CarouselConfig {
  breakpoints?: BreakpointProps[];
  arrowIcon?: Icon;
}

export interface CarouselProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  list: any;
  createElementFunction: CreateElement;
  config: CarouselConfig;
  onVisibleElementsChanged?: Function;
}
