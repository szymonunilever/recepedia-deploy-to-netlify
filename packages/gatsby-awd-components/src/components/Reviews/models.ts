import { RatingAndReviewsProvider, RatingAndReviewsEntityType } from "../../models";

export interface ReviewsProps {
  provider: RatingAndReviewsProvider;
  className?: string;
  linkTo?: string;
  id: number | string;
  entityType: RatingAndReviewsEntityType
}
