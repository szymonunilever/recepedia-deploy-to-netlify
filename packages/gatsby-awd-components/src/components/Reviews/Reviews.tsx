import React, { useState, useEffect } from 'react';
import { ReviewsProps } from './models';
import {
  RatingAndReviewsEntityType,
  ReviewsSummaryTemplate,
  RatingAndReviewsProvider,
} from '../../models';
import cx from 'classnames';
import isBrowser from '../../utils/isBrowser';
import getComponentDataAttrs from '../../utils/getComponentDataAttrs';

const Reviews = ({
  className,
  recipeId,
  provider,
  linkTo = '',
}: ReviewsProps) => {
  const classNames = cx('recipe-reviews', className);

  const [locationOrigin, setLocationOrigin] = useState('');

  useEffect(() => {
    setLocationOrigin(window.location.origin);
  }, []);

  return (
    <>
      {provider === RatingAndReviewsProvider.kritique ? (
        <div
          className={classNames}
          {...getComponentDataAttrs('recipe-reviews')}
        >
          <div
            className="rr-widget-container rr-container"
            data-readpanel-template={ReviewsSummaryTemplate.readpanel01}
            data-entity-type={RatingAndReviewsEntityType.recipe}
            data-unique-id={recipeId}
            data-entity-url={isBrowser() && `${locationOrigin}${linkTo}`}
            data-category-pageurl={isBrowser() && `${locationOrigin}/recipes`}
          />
        </div>
      ) : null}
    </>
  );
};

export default Reviews;
