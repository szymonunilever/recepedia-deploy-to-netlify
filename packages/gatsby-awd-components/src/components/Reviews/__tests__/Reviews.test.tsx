import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import Reviews, { ReviewsProps } from '../index';
import toJson from 'enzyme-to-json';
import { RatingAndReviewsProvider } from '../../../models/ratings&reviews';

describe('<Reviews />', () => {
  let wrapper: ReactWrapper;
  const reviewsProps: ReviewsProps = {
    provider: RatingAndReviewsProvider.kritique,
    className: 'test-class',
    recipeId: 123,
  };

  beforeEach(() => {
    wrapper = mount(<Reviews {...reviewsProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
