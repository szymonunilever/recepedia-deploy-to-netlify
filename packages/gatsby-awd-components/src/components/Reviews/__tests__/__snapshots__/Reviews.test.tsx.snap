// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews /> matches the snapshot 1`] = `
<Reviews
  className="test-class"
  provider={0}
  recipeId={123}
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    script={
      Array [
        Object {
          "async": true,
          "id": "rr-widget-123",
          "src": "undefined?brandid=undefined&localeid=undefined&apikey=undefined&sitesource=undefined",
        },
      ]
    }
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      script={
        Array [
          Object {
            "async": true,
            "id": "rr-widget-123",
            "src": "undefined?brandid=undefined&localeid=undefined&apikey=undefined&sitesource=undefined",
          },
        ]
      }
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        script={
          Array [
            Object {
              "async": true,
              "id": "rr-widget-123",
              "src": "undefined?brandid=undefined&localeid=undefined&apikey=undefined&sitesource=undefined",
            },
          ]
        }
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
  <div
    className="recipe-reviews test-class"
    data-componentname="recipe-reviews"
    data-componentview="recipe-reviews"
  >
    <div
      className="rr-widget-container rr-container"
      data-category-pageurl="http://localhost/recipes"
      data-entity-type="recipe"
      data-entity-url="http://localhost"
      data-readpanel-template="readpanel01"
      data-unique-id={123}
    />
  </div>
</Reviews>
`;
